### PR TITLE
dev/core#1197 Fix event participant calculation to treat 0 count as 0

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
As described in https://lab.civicrm.org/dev/core/-/issues/1197

Before
----------------------------------------
I have 20 participants on a price set
- 1 row has a price option with a `count` of NULL
- 2 rows have price options with a `count` of 0
- 17 rows have price options with a count of 2

Total that shows is 37  - note you need an event AND fee levels for the totalParticipants to show in the participant search

image

After
----------------------------------------
This calculates to 35

Technical Details
----------------------------------------

Comments
----------------------------------------
I will put a note about this in the dev digest - if affects the event full calculation for more purposes than just this one so it needs not to cause a regression & I think there might be some odd data variations out there. I did try creating some rows as multiple participant rows & they seem OK....

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@twomice

**Comment by twomice** [comment]
**TLDR;** This is a complex issue that needs more than just tweaking the "Actual participant count" calculation logic, and this PR causes the calculation to be less accurate in some cases, so we should not merge this.

Thanks to @eileenmcnaughton for the PR, and for helping me analyze this at the sprint.

As I've mentioned [in the ticket](https://lab.civicrm.org/dev/core/-/issues/1197#note_83007), I think there's some ambiguity in the meaning of the `participant_count` property in various situations, and suggest that merely tweaking the counting logic won't resolve that problem. But that conversation is out of scope here, I suppose.

In the context of this PR, it looks like it might improve the count accuracy for some situations, but here's one situation where it definitely makes the count _less_ accurate:

Examine the participant search results below. These participants are using a price set with two fields:
1. "Tournament fees": A radio set with these options:  
fee-options
2. "Extra seat 1" a Text / Numeric Quantity field with Participant Count = 1

We then register a small set of participants with various price selections, and compare the count between master and PR:

On master (without the PR):
count-core

With this PR:
count-pr


With these selections, the "Actual participant count" should be at least 5 (or even more depending on the meaning of "count=0" for Bachman). But with the PR, it says 4, and I can't see any way to get a total of 4 from these participants.

**Comment by eileenmcnaughton** [comment]
@twomice just coming back to the last question at the sprint - we agreed the situation is still a mess after this PR - but is it improved (or even neutral with better code) by this PR?

**Comment by eileenmcnaughton** [comment]
(if not I will close it)

**Comment by twomice** [comment]
@eileenmcnaughton , this PR causes the calculation to be less accurate in some cases, so in my estimation we should not merge this.

**Comment by eileenmcnaughton** [comment]
thanks @twomice
